lib/streams/compound-duplex.js

```diff
@@ -15,6 +15,12 @@ function CompoundDuplex(writable, readable, options) {
 	var self = this;
 	var convertToZStream = require('../index');	// for circular dependencies
 
+	if(!readable && !options && (!writable || typeof writable.write !== 'function')) {
```

Comment:
This argument parsing behavior needs to be documented in the class comment ... it's not necessarily intuitive.